netbox_onboarding/models.py

```diff
@@ -62,3 +75,48 @@ class Meta:  # noqa: D106 "missing docstring in public nested class"
     def __str__(self):
         """String representation of an OnboardingTask."""
         return f"{self.site} : {self.ip_address}"
+
+
+class OnboardingDevice(models.Model):
+    """The status of each Onboarded Device is tracked in the OnboardingDevice table."""
+
+    device = models.OneToOneField(to="dcim.Device", on_delete=models.CASCADE)
+    enabled = models.BooleanField(default=True)
```

Comment:
Mind adding something along the lines of `help_text="Whether (re)onboarding of this device is permitted"`? The meaning of this flag wasn't immediately apparent to me.